### PR TITLE
docs(typehead): add demo for typeaheadSingleWords

### LINKED ISSUE #3843: docs(typehead): add demo for typeaheadSingleWords
If typeaheadSingleWords=true ignores spaces and seeking for matches with occurrences of entered words. 
For example: 
- Ari zona -> Arizona
- A l a b a m a -> Alabama
 

Related to #3342

### PR DESCRIPTION
**Pull request by artolshansky**
If typeaheadSingleWords=true ignores spaces and seeking for matches with occurrences of entered words.
For example:
- Ari zona -> Arizona
- A l a b a m a -> Alabama

Close #3843

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+typeahead/demos/single-world/single-world.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-typeahead-single-world',
+  templateUrl: './single-world.html'
+})
+export class DemoTypeaheadSingleWorldComponent {
+  typeaheadSingleWords = true;
+  selected: string;
+  states: string[] = [
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/demos/single-world/single-world.ts]: remove type annotation

File: demo/src/app/components/+typeahead/typeahead-section.list.ts [hunk 2]
```diff
@@ -70,6 +71,14 @@ export const demoComponentContent: ContentSection[] = [
         html: require('!!raw-loader?lang=markup!./demos/grouping/grouping.html'),
         outlet: DemoTypeaheadGroupingComponent
       },
+      {
+        title: 'Ignore spaces',
+        anchor: 'single-world',
+        component: require('!!raw-loader?lang=typescript!./demos/single-world/single-world.ts'),
+        html: require('!!raw-loader?lang=markup!./demos/single-world/single-world.html'),
+        description: `If <code>typeaheadSingleWords=true</code> ignores spaces and seeking for matches with occurrences of entered words.`,
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/typeahead-section.list.ts]: add example (the one from PR description would fit)

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
As this parameter ignores not only spaces, but order too, will be better rename demo to "Ignore spaces and order", or something like that
As you can see at screenshot, zona Ari -> Arizona
notcorrectname
Otherwise looks good

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Tested, looks good

**Review by valorkin — DISMISSED** [review]

**Pull request merged**